src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -221,32 +232,47 @@ bool InputFileEligibleForCompilation(ITaskItem file)
                     using (var pereader = new PEReader(fs))
                     {
                         if (!pereader.HasMetadata)
-                            return false;
-
-                        if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
-                            return false;
+                        {
+                            return Eligibility.None;
+                        }
 
                         MetadataReader mdReader = pereader.GetMetadataReader();
                         if (!mdReader.IsAssembly)
                         {
-                            return false;
+                            return Eligibility.None;
+                        }
+
+                        if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
+                        {
+                            return Eligibility.ReferenceOnly;
                         }
 
                         // Skip reference assemblies and assemblies that reference winmds
                         if (ReferencesWinMD(mdReader) || IsReferenceAssembly(mdReader) || !HasILCode(pereader, mdReader))
                         {
-                            return false;
+                            return Eligibility.ReferenceOnly;
                         }
                     }
                 }
                 catch (BadImageFormatException)
                 {
                     // Not a valid assembly file
-                    return false;
+                    return Eligibility.None;
                 }
             }
 
-            return true;
+            if (file.HasMetadataValue(MetadataKeys.ReferenceOnly, "true"))
```

Comment:
Just to confirm here, again, we need to make sure that the `ReferenceOnly` metadata does not get added to reference assemblies, since crossgen can't take those as references